cdn/single-origin.yml

```diff
@@ -153,11 +173,34 @@ Parameters:
     Type: String
     AllowedPattern: ^.+\.[a-zA-Z]+$
     ConstraintDescription: >-
-      must be a hostname or of the format mybucket.s3.amazonaws.com
+      must be a hostname or of the format mybucket.s3.region.amazonaws.com
     Description: >-
       The DNS name of the Amazon Simple Storage Service (S3) bucket or the HTTP
       server from which you want CloudFront to get objects for this origin.
-      (e.g. "example.com" or "mybucket.s3.amazonaws.com")
+      (e.g. "example.com" or "mybucket.s3.us-east-2.amazonaws.com")
+  FailoverOriginDomain:
+    Type: String
+    AllowedPattern: ^$|^.+\.[a-zA-Z]+$
+    ConstraintDescription: >-
+      must be a hostname or of the format mybucket.s3.region.amazonaws.com
+    Description: >-
+      (optional) The DNS name of the Amazon Simple Storage Service (S3) bucket
+      or the HTTP server from which you want CloudFront to get objects when the
+      primary origin is unavailable.
+      (e.g. "example.com" or "mybucket.s3.us-east-2.amazonaws.com")
+  FailoverCriteriaStatusCodes:
+    Type: String
+    Description: >-
+      The status codes that, when returned by the primary origin, trigger
+      CloudFront to attempt to use the failover origin.
+    Default: "403-404-500-502-503-504"
+    AllowedValues:
+      # When adding an additional value, a matching entry must be added to the
+      # FailoverCriteriaStatusCodesQuantity mapping which defines the number of
+      # codes present in the value.
+      - "403-404"
```

Comment:
Does this happen before or after status codes are translated by the CDN?  (We usually change 403s to 404s in our distributions... since 403 is what S3 actually returns when an object doesn't exist).

I suppose if you've always got 403/404 grouped together here, it doesn't actually matter.

Reply:
> The primary origin returns an HTTP status code that you’ve configured for failover

I believe failover is based on the actual response code coming back from the origins

Reply:
Also worth noting:

>CloudFront routes all incoming requests to the primary origin, even when a previous request failed over to the secondary origin. CloudFront only sends requests to the secondary origin after a request to the primary origin fails.